packages/picasso.js/src/core/chart-components/axis/axis-arc-label-node.js

```diff
@@ -0,0 +1,102 @@
+function appendStyle(struct, buildOpts) {
+  ['fill', 'fontSize', 'fontFamily'].forEach((style) => {
+    struct[style] = buildOpts.style[style];
+  });
+}
+
+function polarToCartesian(centerX, centerY, radius, angle) {
+  return {
+    x: centerX + radius * Math.cos(angle),
+    y: centerY + radius * Math.sin(angle),
+  };
+}
+
+function checkText(text) {
+  return typeof text === 'string' || typeof text === 'number' ? text : '-';
+}
+
+function collider(struct, tickPos) {
+  if (struct.align === 'right') {
+    if (tickPos.x >= struct.x) {
+      struct.text = '';
+    }
+  }
+}
+
+function calculateMaxWidth(buildOpts, side, innerPos) {
+  let maxWidth;
+  if (side === 'left') {
+    maxWidth = innerPos.x;
+  } else if (side === 'right') {
+    maxWidth = buildOpts.outerRect.width - innerPos.x;
+  } else {
+    maxWidth = Math.max(innerPos.x, buildOpts.outerRect.width - innerPos.x);
+  }
+  return maxWidth;
+}
+
+function appendCollider(tick, struct, buildOpts, tickPos) {
+  collider(tick, struct, buildOpts, tickPos);
+}
+
+function appendBounds(struct, buildOpts) {
+  struct.boundingRect = buildOpts.textBounds(struct);
+}
+
+export default function buildArcLabels(tick, buildOpts) {
+  const rect = buildOpts.innerRect;
+  const centerPoint = { cx: rect.width / 2, cy: rect.height / 2 };
+  const plotSize = Math.min(rect.height, rect.width) / 2;
+  const innerRadius = buildOpts.radius !== undefined ? plotSize * buildOpts.radius : plotSize * 0.5;
```

Comment:
We can simplify it like this:
const innerRadius = plotSize * (buildOpts.radius ?? 0.5);